emission/storage/decorations/trip_queries.py

```diff
@@ -185,17 +218,25 @@ def get_user_input_from_cache_series(user_id, trip_obj, user_input_key):
     potential_candidates = estsc.find_entries(user_id, [user_input_key], tq)
     return final_candidate(valid_user_input(ts, trip_obj), potential_candidates)
 
-def get_additions_for_trip_object(ts, trip_obj):
-    tq = estt.TimeQuery("data.start_ts", trip_obj.data.start_ts, trip_obj.data.end_ts)
-    potential_candidates = ts.find_entries(["manual/trip_addition_input"], tq)
-    return get_not_deleted_candidates(valid_user_input(ts, trip_obj), potential_candidates)
-
-def valid_trip(ts, user_input):
-    def curried(trip_obj):
-        return valid_user_input_for_trip(ts, trip_obj, user_input)
+def get_additions_for_timeline_entry_object(ts, timeline_entry):
+    # timeline_entry can be either a trip or place, so let's check for start/enter or end/exit
+    if (hasattr(timeline_entry.data, 'start_ts')):
+        start = timeline_entry.data.start_ts
+        end = timeline_entry.data.end_ts
+    else:
+        start = timeline_entry.data.enter_ts
+        exit_ts = timeline_entry.data.exit_ts
+        end = exit_ts if exit_ts is not None else EPOCH_MAXIMUM
+    tq = estt.TimeQuery("data.start_ts", start, end)
+    potential_candidates = ts.find_entries(["manual/trip_addition_input", "manual/place_addition_input"], tq)
+    return get_not_deleted_candidates(valid_user_input(ts, timeline_entry), potential_candidates)
+
+def valid_timeline_entry(ts, user_input):
+    def curried(confirmed_obj):
+        return valid_user_input_for_timeline_entry(ts, confirmed_obj, user_input)
     return curried
 
-def get_trip_for_user_input_obj(ts, ui_obj):
+def get_confirmed_obj_for_user_input_obj(ts, ui_obj):
```

Comment:
I thought we know what kind of user input it is (trip input or place input). It seems like it would be more efficient (and likely more correct) to pass that in, and not search through all trips and places.